Bukkit/src/main/java/com/plotsquared/bukkit/listener/BlockEventListener.java

```diff
@@ -324,6 +326,30 @@ public void blockCreate(BlockPlaceEvent event) {
                     return;
                 }
             }
+            // The other block part can be placed outside the plot (#3483)
+            if (event.getBlock().getBlockData() instanceof Bed bed) {
+                org.bukkit.Location other = event.getBlock().getLocation().add(bed.getFacing().getDirection());
+                if (!(Objects.equals(BukkitUtil.adapt(other).getPlot(), plot))) {
+                    pp.sendMessage(
+                            TranslatableCaption.of("permission.no_permission_event"),
+                            Template.of("node", String.valueOf(Permission.PERMISSION_ADMIN_BUILD_ROAD))
+                    );
+                    event.setCancelled(true);
+                    return;
+                }
+            }
+            // Same goes for doors, which may exceed the area height limit
+            if (event.getBlock().getBlockData() instanceof Door door) {
```

Comment:
Assumedly there isn't a utility method for checking the block "size"?

Reply:
Haven't found anything there sadly. There is the Directional interface, but that doesn't indicate that the block contains more than one block state at a time.

Reply:
Just remembered that there's this event https://hub.spigotmc.org/javadocs/spigot/org/bukkit/event/block/BlockMultiPlaceEvent.html